src/prefect/blocks/core.py

```diff
@@ -611,11 +611,16 @@ def _from_block_document(cls, block_document: BlockDocument):
     def _event_kind(self) -> str:
         return f"prefect.block.{self.get_block_type_slug()}"
 
-    def _event_method_called_resources(self) -> ResourceTuple:
-        name = self._block_document_name if self._block_document_name else "anonymous"
+    def _event_method_called_resources(self) -> Optional[ResourceTuple]:
+        if not self._block_document_id:
+            return None
+
         return (
             {
-                "prefect.resource.id": f"prefect.block-document.{name}",
+                "prefect.resource.id": (
+                    f"prefect.block-document.{self._block_document_id}"
+                ),
+                "prefect.block-document.name": self._block_document_name,
```

Comment:
I think we've used `prefect.name` in other cases...I think it's better if we have fewer kinds of labels and use the same label to represent the same concept (like a `prefect.name` or a `prefect.slug`)